src/test/java/picard/analysis/CompareMetricsTest.java

```diff
@@ -0,0 +1,121 @@
+    public void testFailCompareMetricsInvalidMetricToIgnore() {
+        final File input1 = new File(TEST_DATA_DIR, "test1.arrays_variant_calling_detail_metrics");
+        final File input2 = new File(TEST_DATA_DIR, "test2.arrays_variant_calling_detail_metrics");
+        final CompareMetrics compareMetrics = new CompareMetrics();
+        compareMetrics.INPUT = asList(input1, input2);
+        compareMetrics.METRICS_TO_IGNORE = Collections.singletonList("NONEXISTENT_METRIC");
+        Assert.assertEquals(compareMetrics.instanceMain(new String[0]), 1);
+    }
+
+    @Test(expectedExceptions = PicardException.class)
+    public void testFailCompareMetricsInvalidMetricToAllowableRelativeChange() {
+        final File input1 = new File(TEST_DATA_DIR, "test1.arrays_variant_calling_detail_metrics");
+        final File input2 = new File(TEST_DATA_DIR, "test2.arrays_variant_calling_detail_metrics");
+        final CompareMetrics compareMetrics = new CompareMetrics();
+        compareMetrics.INPUT = asList(input1, input2);
+        compareMetrics.METRIC_ALLOWABLE_RELATIVE_CHANGE = Collections.singletonList("NONEXISTENT_METRIC:0.3");
+        Assert.assertEquals(compareMetrics.instanceMain(new String[0]), 1);
+    }
+
+    @DataProvider(name = "testCompareMetricValuesDataProvider")
+    public Object[][] testCompareMetricValuesDataProvider() {
+        return new Object[][]{
+                { 1.0, 1.0, true, "", null },
+                { 1.1, 1.0, false, "Changed by 0.1000000 which is outside of the allowable tolerance of 0.0500000", 0.05 },
+                { 1.1, 1.0, false, "Changed by 0.1000000 which is outside of the allowable tolerance of 0.1000000", 0.1 },
+                { 1.1, 1.0, true, "Changed by 0.1000000 which is within the allowable tolerance of 0.1001000", 0.1001 },
+                { 1.1, 1.0, true, "Changed by 0.1000000 which is within the allowable tolerance of 0.2000000", 0.2 },
+                { 1.0f, 1.0f, true, "", null },
+                { 1.1f, 1.0f, false, "Changed by 0.1000000 which is outside of the allowable tolerance of 0.0500000", 0.05 },
+                { 1.1f, 1.0f, false, "Changed by 0.1000000 which is outside of the allowable tolerance of 0.1000000", 0.1 },
+                { 1.1f, 1.0f, true, "Changed by 0.1000000 which is within the allowable tolerance of 0.1001000", 0.1001 },
+                { 1.1f, 1.0f, true, "Changed by 0.1000000 which is within the allowable tolerance of 0.2000000", 0.2 },
+                { 1, 1, true, "", null },
+                { 1f, 1f, true, "", null }
+        };
+    }
+
+    @Test(dataProvider = "testCompareMetricValuesDataProvider")
+    public void testCompareMetrics(final Object value1, final Object value2,
+                                   final boolean expectedEqual, final String expectedDescription, final Double marcValue) {
+        final CompareMetrics compareMetrics = new CompareMetrics();
+        CompareMetrics.MetricToAllowableRelativeChange = new HashMap<>();
```

Comment:
If you did make this `static` for testing, you should be able to do this instead
```
        compareMetrics.METRIC_ALLOWABLE_RELATIVE_CHANGE = Collections.singletonList("M:" + marcValue);
        compareMetrics.customCommandLineValidation();
```

Reply:
yup.  Yay - more test coverage.